### PR TITLE
MDMP - Support for the next round of breaking changes in the core

### PR DESCRIPTION
**Pull request by AlexKornitzer**
This just brings MDMP inline with the new RBinPlugin API changes and updates the PE files to prevent segfaults.

Regression tests now fail only on the info field, I will update this and submit a PR.

### CODE REVIEW

**Comment by AlexKornitzer** [comment]
Let me just squash the previous two commits.

**Comment by radare** [comment]
gh can squash it too np :)

File: libr/bin/p/bin_mdmp.c [hunk 1]
```diff
@@ -23,14 +23,14 @@ static ut64 baddr(RBinFile *arch) {
...
-static Sdb *get_sdb(RBinObject *o) {
-	struct r_bin_mdmp_obj *bin;
+static Sdb *get_sdb(RBinFile *arch) {
+	struct r_bin_mdmp_obj *obj;
 
-	if (!o) return NULL;
+	if (!arch) return NULL;
 
-	bin = (struct r_bin_mdmp_obj *) o->bin_obj;
+  obj = (struct r_bin_mdmp_obj *)arch->o->bin_obj;
```
Comment (radare) [inline libr/bin/p/bin_mdmp.c]: wrong indent

File: libr/bin/format/mdmp/pe/pe.c [hunk 2]
```diff
@@ -1404,7 +1427,7 @@ static VarFileInfo* Pe_r_bin_pe_parse_var_file_info(struct PE_(r_bin_pe_obj_t)*
 		return NULL;
 	}
 
-	varFileInfo->szKey = (ut16*) malloc (VARFILEINFO_UTF_16_LEN);  //L"VarFileInfo"
+	varFileInfo->szKey = (ut16*) malloc (UT16_ALIGN (VARFILEINFO_UTF_16_LEN ));  //L"VarFileInfo"
```
Comment (radare) [inline libr/bin/format/mdmp/pe/pe.c]: extra space

**Comment by radare** [comment]
lgtm

File: libr/bin/format/mdmp/pe/pe.c (hunk 2 above)
Comment (AlexKornitzer) [inline libr/bin/format/mdmp/pe/pe.c]: Will do but this is a verbatim copy from the core :)

File: libr/bin/p/bin_mdmp.c (hunk 1 above)
Comment (AlexKornitzer) [inline libr/bin/p/bin_mdmp.c]: Crap sorry, forgot to set my vim formatting to r2 style!

**Comment by radare** [comment]
ouch, now its even worst, cant merge this

**Comment by AlexKornitzer** [comment]
Yeah not sure what I have done, bare with me!

**Comment by AlexKornitzer** [comment]
Lets just pretend that all never happened ;)

File: libr/bin/p/bin_mdmp.c [hunk 3]
```diff
@@ -23,14 +23,14 @@ static ut64 baddr(RBinFile *arch) {
...
+	struct r_bin_mdmp_obj *obj;
 
-	if (!o) return NULL;
+	if (!arch) return NULL;
 
-	bin = (struct r_bin_mdmp_obj *) o->bin_obj;
+	obj = (struct r_bin_mdmp_obj *)arch->o->bin_obj;
 
-	if (bin && bin->kv) return bin->kv;
+	if (obj && obj->kv) return obj->kv;
```
Comment (radare) [inline libr/bin/p/bin_mdmp.c]: missing braces

File: libr/bin/p/bin_mdmp.c [hunk 4]
```diff
@@ -23,14 +23,14 @@ static ut64 baddr(RBinFile *arch) {
 	return 0LL;
 }
 
-static Sdb *get_sdb(RBinObject *o) {
-	struct r_bin_mdmp_obj *bin;
+static Sdb *get_sdb(RBinFile *arch) {
+	struct r_bin_mdmp_obj *obj;
 
-	if (!o) return NULL;
+	if (!arch) return NULL;
```
Comment (radare) [inline libr/bin/p/bin_mdmp.c]: add braces

**Comment by AlexKornitzer** [comment]
Think I have put the brackets in the correct place

**Comment by radare** [comment]
<3

**Pull request merged**